src/compiler/utilities.ts

```diff
@@ -2642,7 +2653,7 @@ namespace ts {
     }
 
     export function isPushOrUnshiftIdentifier(node: Identifier) {
-        return node.escapedText === "push" || node.escapedText === "unshift";
+        return isIdentifier(node) && node.escapedText === "push" || node.escapedText === "unshift";

```

Comment:
Isn't `isIdentifier(node)` always `true` here?

Reply:
nice catch. True, the node will always be an identifier, fixed